tests/integrations/raftstore/test_snap.rs

```diff
@@ -1187,3 +1187,40 @@ fn test_leader_step_down_after_requesting_snapshot() {
     std::thread::sleep(election_timeout);
     send_rx.try_recv().unwrap_err();
 }
+
+#[test_case(test_raftstore::new_node_cluster)]
+#[test_case(test_raftstore_v2::new_node_cluster)]
+fn test_node_apply_snapshot_by_or_without_ingest() {
+    for snap_min_ingest_size in [ReadableSize::mb(1), ReadableSize::default()] {
+        let mut cluster = new_cluster(0, 4);
+        cluster.cfg.raft_store.raft_log_gc_tick_interval = ReadableDuration::millis(20);
+        cluster.cfg.raft_store.raft_log_gc_count_limit = Some(2);
+        cluster.cfg.raft_store.merge_max_log_gap = 1;
+        cluster.cfg.raft_store.snap_mgr_gc_tick_interval = ReadableDuration::millis(50);
+        cluster.cfg.server.snap_min_ingest_size = snap_min_ingest_size;
+
+        let pd_client = Arc::clone(&cluster.pd_client);
+        // Disable default max peer count check.
+        pd_client.disable_default_operator();
+        cluster.run();
+
+        // In case of removing leader, let's transfer leader to some node first.
+        cluster.must_transfer_leader(1, new_peer(1, 1));
+        cluster.must_put(b"k1", b"v1");
+        cluster.must_put(b"k2", b"v2");
+        pd_client.must_remove_peer(1, new_peer(4, 4));
+        pd_client.add_peer(1, new_peer(4, 5));
+        let engine4 = cluster.get_engine(4);
+        must_get_equal(&engine4, b"k1", b"v1");
+        must_get_equal(&engine4, b"k2", b"v2");
```

Comment:
Please also check the number of sst files.

Reply:
done.

Reply:
I mean sst files in KV RocksDB.

Reply:
done.